### PR TITLE
KOGITO-8257: Create an integration test for Dashbuilder VSCode Extension

### PR DESCRIPTION
**Pull request by barboras7**

### CODE REVIEW

File: packages/vscode-extension-dashbuilder-editor/package.json [hunk 1]
```diff
@@ -46,12 +49,19 @@
...
     "@types/vscode": "^1.66.0",
     "@vscode/test-web": "^0.0.30",
     "@vscode/vsce": "^2.15.0",
+    "chai": "^4.3.4",
     "copy-webpack-plugin": "^11.0.0",
+    "cpr": "^3.0.1",
     "rimraf": "^3.0.2",
+    "selenium-webdriver": "^4.7.1",
+    "typescript": "^4.4.2",
+    "vscode-extension-tester": "5.2.1",
```
Comment (tomasdavidorg) [inline packages/vscode-extension-dashbuilder-editor/package.json]:
```suggestion
    "vscode-extension-tester": "5.3.0",
```

File: packages/vscode-extension-dashbuilder-editor/package.json [hunk 2]
```diff
@@ -22,6 +22,9 @@
     "lint": "run-script-if --bool \"$(build-env linters.run)\" --then \"kie-tools--eslint ./src\"",
     "pack:prod": "vsce package --githubBranch main --no-dependencies -o ./dist/vscode_extension_dashbuilder_editor_$npm_package_version.vsix",
     "run:webmode": "pnpm vscode-test-web --browserType=chromium --extensionDevelopmentPath=. --version=stable",
+    "test:it": "run-script-if --ignore-errors \"$(build-env integrationTests.ignoreFailures)\" --bool \"$(build-env integrationTests.run)\" --then \"pnpm test:it:clean\" \"cpr it-tests/resources it-tests-tmp/resources\" \"tsc --project tsconfig.it-tests.json --skipLibCheck --sourceMap true\" \"extest setup-and-run --yarn -c max -u -e ./test-resources -o ./it-tests/settings.json out/*.test.js\"",
+    "test:it:clean": "rimraf ./dist-it-tests && rimraf ./test-resources && rimraf ./out && rimraf ./it-tests-tmp && rimraf test-resou && rimraf *.vsix",
```
Comment (tomasdavidorg) [inline packages/vscode-extension-dashbuilder-editor/package.json]: `rimraf test-resou` is it correct?

File: packages/vscode-extension-dashbuilder-editor/it-tests/helpers/VSCodeTestHelper.ts [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+  VSBrowser,
+  WebDriver,
+  WebView,
+  Workbench,
+} from "vscode-extension-tester";
+
+/**
+ * Common test helper class for VSCode extension testing.
+ * Provides common API to work with VSCode test instance.
+ * Allows you  to open folders, files, close open editor.
```
Comment (tomasdavidorg) [inline packages/vscode-extension-dashbuilder-editor/it-tests/helpers/VSCodeTestHelper.ts]:
```suggestion
 * Allows you to open folders, files, close open editor.
```

File: packages/vscode-extension-dashbuilder-editor/it-tests/helpers/VSCodeTestHelper.ts [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+  private browser: VSBrowser;
+
+  /**
+   * WebDriver of currently used VSBrowser.
+   */
+  private driver: WebDriver;
+
+  /**
+   * Handle for workspace section with directory tree and files that
+   * is located in sidebarview. Used to lookup resources and open them.
```
Comment (tomasdavidorg) [inline packages/vscode-extension-dashbuilder-editor/it-tests/helpers/VSCodeTestHelper.ts:58]:
```suggestion
   * are located in sidebarview. Used to lookup resources and open them.
```

File: packages/vscode-extension-dashbuilder-editor/it-tests/helpers/VSCodeTestHelper.ts [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+    this.sidebarView = await control.openView();
+    assert.isTrue(await this.sidebarView.isDisplayed(), "Explorer side bar view was not opened");
+
+    this.workspaceSectionView = await this.sidebarView.getContent().getSection(folderName);
+    return this.sidebarView;
+  };
+
+  /**
+   * Opens Dashbuilder file from a sidebarview. Expects that the sideBarView will be defined and open.
+   * To define sideBarView a folder needs to be openned using openFolder function.
```
Comment (tomasdavidorg) [inline packages/vscode-extension-dashbuilder-editor/it-tests/helpers/VSCodeTestHelper.ts]:
```suggestion
   * To define sideBarView a folder needs to be opened using openFolder function.
```

File: packages/vscode-extension-dashbuilder-editor/it-tests/helpers/VSCodeTestHelper.ts [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+   * Both webviews (text editor and dashbuilder preview) are loaded and returned
+   * in an array on predefined indexes - see returns definition.
+   *
+   * @param fileName name of the file to open
+   * @param fileParentPath optional, use when file is not in root of resources. This is the path of file's parent directory, relative to resources
+   *                       if not used the file will be looked in root of resources.
```
Comment (tomasdavidorg) [inline packages/vscode-extension-dashbuilder-editor/it-tests/helpers/VSCodeTestHelper.ts:101]:
```suggestion
   *                       if it is not used the file will be looked in root of resources.
```

File: packages/vscode-extension-dashbuilder-editor/it-tests/helpers/VSCodeTestHelper.ts [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+   * @returns promise that resolves to an array of WebViews of the opened dashbuilder file.
+   *          The lenght of the array is always 2 and there is guaranteed TextEditor as webview on O index.
```
Comment (tomasdavidorg) [inline packages/vscode-extension-dashbuilder-editor/it-tests/helpers/VSCodeTestHelper.ts]:
```suggestion
   *          The lenght of the array is always 2 and TextEditor is guaranteed as webview on 0 index.
```

File: packages/vscode-extension-dashbuilder-editor/it-tests/helpers/VSCodeTestHelper.ts [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+    for (const notification of activeNotifications) {
+      await notification.dismiss();
+    }
+  };
+
+  /**
+   * Waits until the provided webview has fully loaded kogito editor.
+   * Method will look in the webview for active iframe and switches to the
+   * iframe if located.
+   * After that it looks for div#envelope-app in the iframe#active-frame and if found,
```
Comment (tomasdavidorg) [inline packages/vscode-extension-dashbuilder-editor/it-tests/helpers/VSCodeTestHelper.ts:164]:
```suggestion
   * After that it looks for div#envelope-app in the iframe#active-frame and if it is found,
```

File: packages/vscode-extension-dashbuilder-editor/package.json (hunk 1 above)
Comment (tomasdavidorg) [reply packages/vscode-extension-dashbuilder-editor/package.json]: I think this should be aligned with other versions which are `5.3.0`.

Comment (zdrapela) [reply packages/vscode-extension-dashbuilder-editor/package.json]: Just updated to `5.4.0` #1491 😉

File: packages/vscode-extension-dashbuilder-editor/it-tests/helpers/VSCodeTestHelper.ts (hunk 4 above)
Comment (barboras7) [reply packages/vscode-extension-dashbuilder-editor/it-tests/helpers/VSCodeTestHelper.ts:58]: the "is" actually refers to "workspace section" that's why it is in singular form

File: packages/vscode-extension-dashbuilder-editor/package.json (hunk 2 above)
Comment (barboras7) [reply packages/vscode-extension-dashbuilder-editor/package.json]: I have actually reused the setup we have for the swf vscode extension editor but you are right that this `rimraf test-resou` has no effect on the result, hence I deleted it.

File: packages/vscode-extension-dashbuilder-editor/package.json [hunk 9]
```diff
@@ -22,6 +22,9 @@
     "lint": "run-script-if --bool \"$(build-env linters.run)\" --then \"kie-tools--eslint ./src\"",
     "pack:prod": "vsce package --githubBranch main --no-dependencies -o ./dist/vscode_extension_dashbuilder_editor_$npm_package_version.vsix",
     "run:webmode": "pnpm vscode-test-web --browserType=chromium --extensionDevelopmentPath=. --version=stable",
+    "test:it": "run-script-if --ignore-errors \"$(build-env integrationTests.ignoreFailures)\" --bool \"$(build-env integrationTests.run)\" --then \"pnpm test:it:clean\" \"cpr it-tests/resources it-tests-tmp/resources\" \"tsc --project tsconfig.it-tests.json --skipLibCheck --sourceMap true\" \"extest setup-and-run --yarn -c max -u -e ./test-resources -o ./it-tests/settings.json out/*.test.js\"",
```
Comment (tomasdavidorg) [inline packages/vscode-extension-dashbuilder-editor/package.json:25]: The tests were not executed. I think we forgot to add ` && pnpm test:it` in the `"build:prod"` script.

**Review by tomasdavidorg — CHANGES_REQUESTED** [review]

**Review by tomasdavidorg — APPROVED** [review]

**Comment by barboras7** [comment]
@tiagobento I am not able to remove the "Do not merge" label. Could you please remove it and look at the PR/tell me who I should ask? 

Also, if removing labels is somehow disabled for this repo, could you or some of the other owners take a look and fix such setting? It's actually quite strange that I am able to add a label but I am not able to remove it afterwards :slightly_smiling_face:

**Comment by caponetto** [comment]
@barboras7 can you please try again?

**Review by caponetto — APPROVED** [review]

**Pull request merged**